src/Syncs/IBLTSync_Adaptive.cpp

```diff
@@ -0,0 +1,254 @@
+//
+// Created by Xingyu Chen on 4/27/25.
+//
+
+#include <GenSync/Syncs/IBLTSync_Adaptive.h>
+#include <GenSync/Aux/Exceptions.h>
+
+IBLTSync_Adaptive::IBLTSync_Adaptive(size_t initExpected, size_t eltSize) {
+    initExpNumElems = initExpected;
+    elementSize = eltSize;
+
+    myIBLT = IBLT::Builder().
+            setNumHashes(4).
+            setNumHashCheck(11).
+            setExpectedNumEntries(initExpNumElems).
+            setValueSize(elementSize).
+            build();
+}
+
+IBLTSync_Adaptive::~IBLTSync_Adaptive() = default;
+
+bool IBLTSync_Adaptive::SyncClient(const shared_ptr<Communicant>& commSync,
+                                   list<shared_ptr<DataObject>> &selfMinusOther,
+                                   list<shared_ptr<DataObject>> &otherMinusSelf) {
+    Logger::gLog(Logger::METHOD, "Entering IBLTSync_Adaptive::SyncClient");
+
+    size_t currentExpected = initExpNumElems;
+
+    // call parent method for bookkeeping
+    SyncMethod::SyncClient(commSync, selfMinusOther, otherMinusSelf);
+
+    // connect to server
+    mySyncStats.timerStart(SyncStats::IDLE_TIME);
+    commSync->commConnect();
+    mySyncStats.timerEnd(SyncStats::IDLE_TIME);
+
+    // keep running until peeling succeed
+    while (true) {
+        commSync->commSend(static_cast<int>(currentExpected));
+
+        // construct new IBLT with updated size
+        myIBLT = IBLT::Builder()
```

Comment:
This is the same code as before ... it should probably be relegated to a helper function.